src/main/java/uk/gov/companieshouse/accounts/association/controller/UserCompanyAssociations.java

```diff
@@ -153,7 +153,7 @@ public ResponseEntity<ResponseBodyPost> inviteUser(final String xRequestId, fina
         LOG.debugContext( xRequestId, String.format( "Attempting to search for %s in accounts-user-api.", inviteeEmail ), null );
         final var inviteeUserDetails = usersService.searchUserDetails( List.of( inviteeEmail ) );
 
-        final var inviteeUserFound = !inviteeUserDetails.isEmpty();
+        final var inviteeUserFound = !( Objects.isNull( inviteeUserDetails ) || inviteeUserDetails.isEmpty() );
```

Comment:
the above needs testing for 8-9 different scenarios.

if you have Objects.NonNull && invitedUserDetails.isPresent then its only 2 scenarios.

Reply:
Good point. I have updated the code.

Since inviteeUserDetails is a list (not an optional), I had to use it's isEmpty method. The code looks like this now:

        final var inviteeUserFound = Objects.nonNull(inviteeUserDetails) && !inviteeUserDetails.isEmpty();